### ISSUE TITLE
Expand Reference Client and Optimizer to reflect all existing functionality

### ISSUE THREAD

**Issue by Groostav** [EmpowerOperations/volition#1]
> As an external programmer looking to use the OASIS API, I want to read (and possibly copy-paste-modify) source code that shows me how to consume the optimization API, _including a working optimizer_ so that I might integrate my simulator with the volition API and OASIS. 

> As a software architect I want a GUI'd reference client and functioning optimizer example so that we might better understand the requirements OASIS will have to support the API.

Once this issue is finished, we should be able to demo simulator code running against the published OASIS API.

This demo should include:
1. starting & connecting multiple simulators 
1. starting an optimization from the simulator
2. starting an optimization from the ref optimizer
3. importing a problem into the ref optimizer, including auto-configuration of variable names and default bounds
4. running successful iterations
5. the simulator emitting a status update message
6. the optimizer requesting a status update message
5. stopping an optimization from the optimizer

This demo will not include any kind of error based path, meaning all code paths may assume success and ignore support for error messages. 

> It should be noted that the plan will be to push the `XYZResponse` and `ABCConfirm` message types to have a `one-of` field to express their result, where the result will be `one-of` a success case or an enumerated set of error cases. If, in doing this work, it becomes apparent that this strategy is not possibly or would be excessively difficult we should punt that work out of scope.
